autocomplete/builders/common.lua

```diff
@@ -393,54 +408,102 @@ end
 -- Package compilation
 --
 
+---@alias packageType
+---|"class"
+---|"function"
+---|"method"
+---|"value"
+---|"lib"
+---|"event"
+
 --- @class exampleTable
 --- @field title string|nil The example title.
 --- @field description string|nil The description of the example.
 
 --- @class package
+--- @field name string
+--- @field useDefault boolean
+--- @field readOnly boolean
+--- @field optional boolean? Is this parameter optional?
+--- @field default string|integer|nil
+--- @field experimental boolean
+--- @field description string?
 --- @field key string The name of the file that generated this package.
---- @field type string The type definition for the package.
+--- @field type packageType The type definition for the package.
 --- @field folder string The folder that the package was created from.
 --- @field parent package The package this package is a child of.
 --- @field namespace string The full namespace of the package.
 --- @field deprecated boolean Allows marking definitions as deprecated. Those definitions aren't written to the web documentation.
 --- @field examples table<string, exampleTable>|nil A table containing the examples. Keys are the example's name/path to the example file.
 
+---@class packageValue : package
+---@field valuetype string
+---@field link string
+
+---@class packageFunctionArgument : package
+---@field tableParams packageFunctionArgument[]
+
+--- @class packageFunction : packageValue
+--- @field arguments packageFunctionArgument[]
+--- @field returns packageFunctionArgument[]|nil
+
+--- @class packageMethod : packageFunction
+
 --- @class packageLib : package
 --- @field children table<string, package>|nil
---- @field functions package[]|nil
---- @field values package[]|nil
+--- @field functions packageFunction[]|nil
+--- @field values packageValue[]|nil
+
+
+---@class packageOverload
+---@field rightType string
+---@field resultType string
+---@field description string
+
+---@class packageOperator : package
+---@field overloads packageOverload[]
 
 --- @class packageClass : packageLib
 --- @field inherits string The class that this class descends from.
+--- @field operators packageOperator[]
 --- @field isAbstract boolean
---- @field methods package[]|nil
+--- @field methods packageMethod[]|nil
 --- @field allDescendents table<string, packageClass>
 --- @field directDescendents table<string, packageClass>
+--- @field allDescendentKeys string
 
---- @class packageFunction : package
 
---- @class packageMethod : packageFunction
---- @field returns table
+
+---@class packageEvent : package
+---@field filter string
+---@field blockable boolean
+---@field related string[]
+---@field eventData table<string, packageFunctionArgument>
```

Comment:
This one can be linked to: https://github.com/MWSE/MWSE/blob/master/docs/event-definitions-guide.md